### ISSUE TITLE
Improving documentation for OpenAPI support

### ISSUE THREAD

**Issue by patrickoconnor-wf** [specmatic/specmatic#562]
**Is your feature request related to a problem? Please describe.**
I'm frustrated with the documentation related to OpenAPI support.

**Describe the solution you'd like**
I've read through the [documentation site](https://specmatic.in/documentation/) but I'm struggling with how to actually use the project.

For context, I have an OpenAPI spec. This is an existing API that I'd like to be able to do contract testing with. How do I actually provide an `Authorization` header when running the test on my spec? Specifically, `java -jar specmatic.jar test mySpec.yaml`. The documentation says that `bearer` and `apiKey` are supported but I can't seem to find anything in the documentation that would allow me to specify what they should be.

**Describe alternatives you've considered**
Is the intended way to use this project to write separate contracts that mirror what already exists in the spec? I figured out that I could write a contract like this with the `Authorization` header assuming I store the value of the header in a file called `token.txt`:
```
Feature: Documents
    Scenario Outline: Get Documents
        When GET /documents
        And request-header Authorization (string)
        Then status 200

        Examples:
            | Authorization|
            | (@token.txt)|
```

In summary, I feel like I should be able to use my existing spec with filled-out examples as a way to run tests with specmatic, but I haven't been able to connect the dots with the existing documentation on how to actually provide authorization for my requests.

**Additional context**
I think this project would be a perfect fit for what I'm trying to do. I really appreciate all the work you guys have done to build such a cool tool. Thanks!

**Comment by harikrishnan83** [comment]
@patrickoconnor-wf Thanks a lot for your detailed feedback. We really appreciate it.

I have updated below sections in our documentation.
* [**Getting started**](https://specmatic.in/getting_started.html) - Includes links to video demos
* [**Authentication**](https://specmatic.in/documentation/authentication.html)

Please do take a look and let me know if you see any missing pieces.

While [we do not recommend having real auth wired up for running "Contract as Test"](https://specmatic.in/documentation/authentication.html), we are working on a mechanism to provide auth tokens for running contract as tests for corner cases, will keep you posted on this.

Also we would be happy get on a quick call to understand your context better. Please feel free to reach us at [specmatic.in](https://specmatic.in/contact-us/) to setup a call with us.

**Comment by patrickoconnor-wf** [comment]
Thanks a lot! Relying on a mock auth implementation is not something I had considered. That seems like something we could work with. I'll take a closer look at these additions and get back to you. We're exploring some options for contract testing internally so we might set up that call. 

Thanks again for all the work you've put into this :partying_face:

**Comment by harikrishnan83** [comment]
@patrickoconnor-wf Hope you are doing well. We have some recent updates to the authentication and now support ability to set the real bearer tokens, API keys, etc. through env variables. Here is the [documentation](https://specmatic.in/documentation/authentication.html#testing-with-real-auth).

**Comment by patrickoconnor-wf** [comment]
Thank you! We'll take a look. Happy new year! 🎉
